harness/determined/cli/user_groups.py

```diff
@@ -0,0 +1,262 @@
+    for user_id, username in zip(user_ids, usernames):
+        print(f"user removed from the group with username {username} and ID {user_id}")
+
+
+@authentication.required
+@require_feature_flag("rbacEnabled", rbac_flag_disabled_message)
+def change_group_name(args: Namespace) -> None:
+    session = setup_session(args)
+    group_id = group_name_to_group_id(session, args.old_group_name)
+    body = bindings.v1UpdateGroupRequest(groupId=group_id, name=args.new_group_name)
+    resp = bindings.put_UpdateGroup(session, groupId=group_id, body=body)
+    g = resp.group
+
+    print(f"user group with ID {g.groupId} name changed from {args.old_group_name} to {g.name}")
+
+
+@authentication.required
+@require_feature_flag("rbacEnabled", rbac_flag_disabled_message)
+def delete_group(args: Namespace) -> None:
+    if args.yes or render.yes_or_no(
+        "Deleting a group will result in an unrecoverable \n"
+        "deletion of the group along with all the membership  \n"
+        "information of the group. Do you still wish to proceed? \n"
+    ):
+        session = setup_session(args)
+        group_id = group_name_to_group_id(session, args.group_name)
+        bindings.delete_DeleteGroup(session, groupId=group_id)
+        print(f"user group with name {args.group_name} and ID {group_id} deleted")
+    else:
+        print("Skipping group deletion.")
+
+
+def usernames_to_user_ids(session: session.Session, usernames: List[str]) -> List[int]:
+    usernames_to_ids: Dict[str, Optional[int]] = {u: None for u in usernames}
+    users = bindings.get_GetUsers(session).users or []
+    for user in users:
+        if user.username in usernames_to_ids:
+            usernames_to_ids[user.username] = user.id
+
+    missing_users = []
+    user_ids = []
+    for username, user_id in usernames_to_ids.items():
+        if user_id is None:
+            missing_users.append(username)
+        else:
+            user_ids.append(user_id)
+
+    if missing_users:
+        raise api.errors.BadRequestException(
+            f"could not find users for usernames {', '.join(missing_users)}"
+        )
+    return user_ids
+
+
+def group_name_to_group_id(session: session.Session, group_name: str) -> int:
+    body = bindings.v1GetGroupsRequest(name=group_name, limit=1, offset=0)
+    resp = bindings.post_GetGroups(session, body=body)
+    groups = resp.groups
+    if groups is None or len(groups) != 1 or groups[0].group.groupId is None:
+        raise api.errors.BadRequestException(f"could not find user group name {group_name}")
+    return groups[0].group.groupId
+
+
+args_description = [
+    Cmd(
+        "user-group",
+        None,
+        "manage user groups",
+        [
+            Cmd(
+                "create",
+                create_group,
+                "creates a user group",
+                [
+                    Arg("group_name", default=None, help="name of user group to be created"),
```

Comment:
since it's required, maybe shouldn't default to None?
```suggestion
                    Arg("group_name", help="name of user group to be created"),
```

(this should be applied to all required positional arguments here)